lib/csv_row_model/coercer.rb

```diff
@@ -1,32 +0,0 @@
-module CsvRowModel
-  class Coercer < Struct.new(:options, :caller)
-
-    def decode(value)
-      caller.instance_exec(value, &parse_lambda)
-    end
-
-    private
-
-    # @return [Lambda, Proc] returns the Lambda/Proc given in the parse option or:
-    # ->(original_value) { parse_proc_exists? ? parsed_value : original_value  }
-    def parse_lambda
-      raise ArgumentError.new("You need either :parse OR :type but not both of them") if options[:parse] && options[:type]
-      parse_lambda = options[:parse] || CLASS_TO_PARSE_LAMBDA[options[:type]]
-      return parse_lambda if parse_lambda
-      raise ArgumentError.new("type must be #{CLASS_TO_PARSE_LAMBDA.keys.reject{|e|e.nil?}.join(', ')}")
-    end
-
-    # Mapping of column type classes to a parsing lambda. These are applied after {Import.format_cell}.
-    # Can pass custom Proc with :parse option.
-    CLASS_TO_PARSE_LAMBDA = {
-      nil => ->(s) { s },
-      # inspired by https://github.com/MrJoy/to_bool/blob/5c9ed38e47c638725e33530ea1a8aec96281af20/lib/to_bool.rb#L23
-      Boolean => ->(s) { s =~ /^(false|f|no|n|0|)$/i ? false : true },
-      String  => ->(s) { s },
-      Integer => ->(s) { s.to_i },
-      Float   => ->(s) { s.to_f },
-      Date    => ->(s) { s.present? ? Date.parse(s) : s }
-    }
-
-  end
-end
```

Comment:
@joel sorry that i removed this..... i thought about it..... the dependencies made me feel like i should change it. what i mean is... this class has 3 dependencies: `options[:type], options[:parse], caller`:

- `caller` is only required in 1 line.... so why have this here
- `options` is from `Import`... if `Import` changes the name of `:type` or `:parse`, then we have to change it here, it isn't obvious....

So... to make it more independent, i think it's better like this:
```ruby
module CsvRowModel
  class Coercer < Struct.new(:parse, :type)
    # @return [Lambda, Proc] returns the Lambda/Proc given in the parse option or:
    # ->(original_value) { parse_proc_exists? ? parsed_value : original_value  }
    def parse_lambda
      raise ArgumentError.new("You need either :parse OR :type but not both of them") if parse && type
      parse_lambda = parse || CLASS_TO_PARSE_LAMBDA[type]
      return parse_lambda if parse_lambda
      raise ArgumentError.new("type must be #{CLASS_TO_PARSE_LAMBDA.keys.reject{|e|e.nil?}.join(', ')}")
    end

    CLASS_TO_PARSE_LAMBDA = {
      nil => ->(s) { s },
     .....
    }
  end
end
```

but... `"You need either :parse OR :type but not both of them"` implies that `:parse` and `:type` is an option too.

you kinda see what i mean...?

If it's an independent class like a validator... it makes total sense to me. i took this out to the `Attributes` module. it seemed like a good grouping.

i felt bad changing it -_-